File: backend/src/main/java/com/back/team11/domain/member/dto/MemberResponseDto.java
package com.back.team11.domain.member.dto;

import com.back.team11.domain.member.entity.Member;

public record MemberResponseDto(
        Long memberId,
        String nickname,
        String email,
        String role
) {
    public static MemberResponseDto from(Member member) {
        return new MemberResponseDto(
                member.getId(),
                member.getNickname(),
                member.getEmail(),
                member.getRole().name()
        );
    }
}

File: backend/src/main/java/com/back/team11/domain/member/dto/MemberResponseDto.kt
package com.back.team11.domain.member.dto

import com.back.team11.domain.member.entity.Member

data class MemberResponseDto(
    val memberId: Long,
    val nickname: String,
    val email: String,
    val role: String
) {
    companion object {
        fun from(member: Member): MemberResponseDto = MemberResponseDto(
            memberId = member.id,
            nickname = member.nickname,
            email = member.email,
            role = member.role.name
        )
    }
}
